template.go

```diff
@@ -393,10 +381,52 @@ func generateFile(config Config, containers Context) bool {
 			if err != nil {
 				log.Fatalf("unable to create dest file %s: %s\n", config.Dest, err)
 			}
-			log.Printf("Generated '%s' from %d containers", config.Dest, len(filteredContainers))
+			log.Printf("Generated '%s' from %d out of %d containers.", config.Dest, len(filteredContainers), len(containers))
 			return true
 		}
 		return false
 	}
 	return true
 }
+
+/*
+The idea behind this function is to easily remove a RuntimeContainer pointer
+from the Context slice.
+
+This is intended to be used with a for loop and to pass the "index" as a pointer
+so that after we remove the particular RuntimeContainer pointer, we can bump
+back the index so we don't skip over any of the RuntimeContainer pointers.
+*/
+func removeFilteredContainer(containers *Context, index *int) {
```

Comment:
This is seems more complicated than necessary.  I'd rather the filter functions just return a new filtered slice as opposed to modifying the parameter pointers.